core/zoe/zoe.chainmail

```diff
@@ -0,0 +1,165 @@
+/**
+ * This is returned by a call to `makeInstance` on Zoe.
+ */
+struct UserFacetAndGoverningContract ( ) {
+  zoeInstance :ZoeUserFacet;
+  governingContract :CustomGoverningContract;
+}
+
+/**
+ * This is returned by a call to `escrow` on the zoeInstance userFacet.
+ * An escrowReceipt is an ERTP payment which acts as proof of escrow
+ * and is normally passed in to a governingContract method. A
+ * claimPayoff ERTP payment is what can be unwrapped to claim the
+ * payoff of involvement with a contract. 
+ */
+struct EscrowReceiptAndClaimPayoff ( ) {
+  escrowReceipt :Payment;
+  claimPayoff :Payment;
+}
+
+/**
+ * Zoe is a long-lived and well-trusted contract that enforces
+ * offer-safety for the contracts that use it. Zoe has a single
+ * `seatIssuer` and a single `escrowReceiptIssuer` for the entirety of its
+ * lifetime. By having a reference to Zoe, a user can get the `seatIssuer`
+ * or `escrowReceiptIssuer` and thus validate any `seat` or `escrowReceipt`
+ * they receive from someone else. 
+ */
+
+interface Zoe {
+  /** 
+   * Zoe is long-lived, but we want to create individual zoeInstances
```

Comment:
The word "instance" here is confusing. The long-lived credible thing that Alice and Bob both know before they know each other is indeed an instance of this interface. That instance has a method, currently called `makeInstance`, that is making something specific to a given contract that Alice and Bob will participate it. Let's call this a Zoe Pool. In the previous scenario, Alice creates a Zoe pool and communicates something specific to this pool to Bob for Bob to participate. Bob need to validate that something against the long lived Zoe instance, i.e., the object that implements the `Zoe` interface.